server/src/main/java/org/elasticsearch/index/fielddata/plain/DocValuesIndexFieldData.java

```diff
@@ -71,6 +72,7 @@ public final Index index() {
 
         private NumericType numericType;
         private Function<SortedSetDocValues, ScriptDocValues<?>> scriptFunction = AbstractAtomicOrdinalsFieldData.DEFAULT_SCRIPT_FUNCTION;
+        private RangeType rangeType;
```

Comment:
This is currently only being used as a flag, but the overhead between storing a boolean and storing an enum reference isn't high, so it seemed worth leaving ourselves access to the more robust data.